frontend/components/home/UserCard.jsx

```diff
@@ -1,39 +1,185 @@
+} from 'react-native';
 import { Text, Card, Button, Icon, Divider } from '@rneui/themed';
-const { useState, useEffect, useRef } = React;
+import { getDistance } from 'geolib';
+
 import userData from '../home/exampleData/userData.js';
 import MoreInfo from '../home/MoreInfo.jsx';
+import ImageGallery from '../home/ImageGallery.jsx';
 
+const { useState, useEffect, useRef } = React;
 const dWidth = Dimensions.get('window').width;
 const dHeight = Dimensions.get('window').height;
 
-const UserCard = ({ item, index, onSwipe }) => {
+const UserCard = ({ item, index, handleSwipe, omitCard }) => {
   const [viewMore, setViewMore] = useState(false);
+  const [imgHeight, setImgHeight] = useState(dHeight * 0.58);
+  const [xPosition, setXPosition] = useState(new Animated.Value(0));
+  const [imgIndex, setImgIndex] = useState(0);
+  let swipeDirection = '';
+  let cardOpacity = new Animated.Value(1);
+  let rotateCard = xPosition.interpolate({
+    inputRange: [-200, 0, 200],
+    outputRange: ['-20deg', '0deg', '20deg'],
+  });
+  const panResponder = PanResponder.create({
+    onStartShouldSetPanResponder: (e, gestureState) => false,
+    onMoveShouldSetPanResponder: (e, gestureState) => true,
+    onStartShouldSetPanResponderCapture: (e, gestureState) => false,
+    onMoveShouldSetPanResponderCapture: (e, gestureState) => true,
+    onPanResponderMove: (e, gestureState) => {
+      xPosition.setValue(gestureState.dx);
+      if (gestureState.dx > dWidth - 250) {
+        swipeDirection = 'right';
+      } else if (gestureState.dx < -dWidth + 250) {
+        swipeDirection = 'left';
+      }
+    },
+    onPanResponderRelease: (e, gestureState) => {
+      if (gestureState.dx < dWidth - 150 && gestureState.dx > -dWidth + 150) {
+        handleSwipe('');
+        Animated.spring(xPosition, {
+          toValue: 0,
+          speed: 5,
+          bounciness: 10,
+          useNativeDriver: false,
+        }).start();
+      } else if (gestureState.dx > dWidth - 150) {
+        Animated.parallel([
+          Animated.timing(xPosition, {
+            toValue: dWidth,
+            duration: 200,
+            useNativeDriver: false,
+          }),
+          Animated.timing(cardOpacity, {
+            toValue: 0,
+            duration: 200,
+            useNativeDriver: false,
+          }),
+        ]).start(() => {
+          handleSwipe(swipeDirection, item.id);
+          omitCard();
+        });
+      } else if (gestureState.dx < -dWidth + 150) {
+        Animated.parallel([
+          Animated.timing(xPosition, {
+            toValue: -dWidth,
+            duration: 200,
+            useNativeDriver: false,
+          }),
+          Animated.timing(cardOpacity, {
+            toValue: 0,
+            duration: 200,
+            useNativeDriver: false,
+          }),
+        ]).start(() => {
+          handleSwipe(swipeDirection);
+          omitCard();
+        });
+      }
+    },
+  });
+
+  const calcDistance = (currItem) => {
```

Comment:
Do you want me to include distance in the `GET: /home` route? I already have to calculate it for filtering, so it wouldn't take any extra effort

Reply:
That would be amazing Kate! Processing that on the backend should speed up things on the front-end significantly. I appreciate it, thanks so much.